### PR TITLE
[fix](multi-table-load) fix single stream multi table load can not finish

### PR DESCRIPTION
**Pull request by sollhui**
## Proposed changes

It is necessary to determine whether the sum of pipe current capacity and size is greater than pipe max capacity, otherwise the following situation may occur:
the pipe is full but still cannot trigger the request and exec plan condition, causing one stream multi table load can not finish.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by HHoflittlefish777** [comment]
run buildall

**Comment by HHoflittlefish777** [comment]
run buildall

**Comment by HHoflittlefish777** [comment]
run buildall

**Comment by HHoflittlefish777** [comment]
run buildall

**Review by liaoxin01 — APPROVED** [review]
LGTM

File: be/src/io/fs/multi_table_pipe.cpp [hunk 1]
```diff
@@ -114,25 +112,35 @@ Status MultiTablePipe::dispatch(const std::string& table, const char* data, size
...
-            }
+    }
+
+    // It is necessary to determine whether the sum of pipe_current_capacity and size is greater than pipe_max_capacity,
+    // otherwise the following situation may occur:
+    // the pipe is full but still cannot trigger the request and exec plan condition,
+    // causing one stream multi table load can not finish
+    auto pipe_current_capacity = pipe->current_capacity();
+    auto pipe_max_capacity = pipe->max_capacity();
+    if (_unplanned_row_cnt >= _row_threshold || _unplanned_pipes.size() >= _wait_tables_threshold ||
```
Comment (liaoxin01) [inline be/src/io/fs/multi_table_pipe.cpp:124]: This code needs to be placed in the else section above？

**Pull request closed without merging**